### PR TITLE
Migrated finetuning and inference code of workflow chatbot to use NeuralChat API

### PR DESCRIPTION
**Pull request by XinyuYe-Intel**
## Type of Change

example migrate
API not changed

## Description

Migrated finetuning and inference code of workflow chatbot to use NeuralChat API.
JIRA ticket: xxx

## Expected Behavior & Potential Risk

pass local test

## How has this PR been tested?

follow readme

## Dependency Change?

no library dependency introduced or removed

### CODE REVIEW

**Comment by hshen14** [comment]
@changwangss @lvliang-intel

**Review by hshen14 — APPROVED** [review]

File: intel_extension_for_transformers/neural_chat/docker/Dockerfile [hunk 1]
```diff
@@ -29,7 +29,7 @@
 ARG UBUNTU_VER=22.04
 FROM ubuntu:${UBUNTU_VER} as cpu
 
-ARG ITREX_VER=main
```
Comment (hshen14) [inline intel_extension_for_transformers/neural_chat/docker/Dockerfile:32]: please fix this.

File: intel_extension_for_transformers/neural_chat/docker/Dockerfile [hunk 2]
```diff
@@ -113,7 +113,7 @@ RUN git clone https://github.com/huggingface/optimum-habana.git && \
     pip install -e .
 
 # Download ITREX code
-ARG ITREX_VER=main
```
Comment (hshen14) [inline intel_extension_for_transformers/neural_chat/docker/Dockerfile:116]: same

**Comment by hshen14** [comment]
@XinyuYe-Intel  please fix the remaining issues about the hard coded branch name.

**Comment by XinyuYe-Intel** [comment]
> @XinyuYe-Intel please fix the remaining issues about the hard coded branch name.

Done, it's for test only, now that test have passed, I've changed back.

**Pull request merged**